crypto/include/internal/rand_int.h

```diff
@@ -39,8 +39,15 @@ size_t rand_drbg_get_entropy(RAND_DRBG *drbg,
                              int prediction_resistance);
 void rand_drbg_cleanup_entropy(RAND_DRBG *drbg,
                                unsigned char *out, size_t outlen);
+size_t rand_drbg_get_nonce(RAND_DRBG *drbg,
+                           unsigned char **pout,
+                           int entropy, size_t min_len, size_t max_len);
```

Comment:
copy & paste error :wink:

Reply:
The `min_len` is also not necessary. In that point I will follow Kurt, see #5909.

Reply:
Oh, wait! The  `entropy` argument is there since FIPS 2.0. I will have to check that first...

```C
    int FIPS_drbg_set_callbacks(DRBG_CTX * dctx,
                                size_t (*get_entropy) (DRBG_CTX * ctx,
                                                       unsigned char **pout,
                                                       int entropy,
                                                       size_t min_len,
                                                       size_t max_len),
                                void (*cleanup_entropy) (DRBG_CTX * ctx,
                                                         unsigned char *out,
                                                         size_t olen),
                                size_t entropy_blocklen,
                                size_t (*get_nonce) (DRBG_CTX * ctx,
                                                     unsigned char **pout,
                                                     int entropy,
                                                     size_t min_len,
                                                     size_t max_len),
                                void (*cleanup_nonce) (DRBG_CTX * ctx,
                                                       unsigned char *out,
                                                       size_t olen));
```

Reply:
The parameter makes sense for variant a. And it makes also sense to change the name to `security_strength` as Kurt did [here](https://github.com/openssl/openssl/pull/5909/files#diff-2fea19f1443a1415d4686e753ac201d1R40).

Reply:
It makes sense for variant a, sure...  but this PR is to implement variant b, is it not?

Reply:
For us the parameter is not necessary. But if you view the `RAND_DRBG` "class" as a general purpose building block and not just how it is wired into our RAND_METHOD, then the parameter _does_ make sense. The callbacks (which are present since FIPS 2.0) give the developer full control over the reseeding process. And if the developer provides a get_nonce() callback he needs to know the security level.

That we have a fallback for missing get_entropy callbacks is the consequence of a misunderstanding (we thought they were optional) and a convenience: Retrieving entropy + random_nonce in a single call is  more efficient than to make two generate calls, one in get_entropy() and one in get_nonce().

Reply:
Okie